src/Resources/app/storefront/test/filter-category-select.plugin.test.js

```diff
@@ -155,36 +163,49 @@ describe('filter-category-select.plugin.js', () => {
     });
 
     test('Women and hats categories to be selected', () => {
-        $('#Women_Hats').click();
+        let checkbox = document.querySelector('#Women_Hats');
+        const clickEvent = new Event('click');
+        checkbox.addEventListener('click', ()=>{
+            checkbox.checked = !checkbox.checked;
+            checkbox.dispatchEvent(new Event('change'));
+        });
+        checkbox.dispatchEvent(clickEvent);
+
         let isCategorySelected = document.getElementById('Women').checked;
-        let womenHat = document.getElementById('Women_Hats').checked;
-        let isWomenAddedToUrl = window.location.search.indexOf('Women') > -1;
-        let isWomenHatAddedToUrl = window.location.search.indexOf('Women_Hats') > -1;
+        let isWomenHat = checkbox.checked;
+        let params = window.location.search;
+        let isUrlUpdated = params.indexOf('Women') > -1 && params.indexOf('Women_Hats') > -1;
         expect(isCategorySelected).toBe(true);
-        expect(womenHat).toBe(true);
-        expect(isWomenAddedToUrl).toBe(true);
-        expect(isWomenHatAddedToUrl).toBe(true);
+        expect(isWomenHat).toBe(true);
+        expect(isUrlUpdated).toBe(true);
     });
 
     test('Seleting Men category, click on an icon to hide sub categories', () => {
 
-        let menSub = $('#Men').parent()[0].querySelector('.sub-item')
-        const subCategory = menSub
-        const icon = $('.category-toggle-icon')[0];
-        icon.dispatchEvent(new Event('click'))
-        let isSubHide = subCategory.classList.contains('hide-category-list-item');
-        expect(isSubHide).toBe(true);
+        let men = document.querySelector('#Men');
+        let child = men.parentNode.querySelector('.sub-item')
+        const subCategory = child;
+        const icon = document.querySelector('.category-toggle-icon');
+        icon.dispatchEvent(new Event('click'));
+        let isChildHide = subCategory.classList.contains('hide-category-list-item');
+        expect(isChildHide).toBe(true);
     });
 
     test('Ensure that deselecting an already selected category, will automatically hide all subcategories', () => {
-        const icon = $('.category-toggle-icon')
-        icon[0].dispatchEvent(new Event('click')); // To show Subcategories, so can be hidden on deslecting parent
-        let subCategory = $('#Men').parent()[0].querySelector('.sub-item');
-        let isSubHide = subCategory.classList.contains('show-category-list-item');
-        expect(isSubHide).toBe(true);
-        $('#Men').click();
-        isSubHide = subCategory.classList.contains('hide-category-list-item');
-        expect(isSubHide).toBe(true);
+        const icon = document.querySelector('.category-toggle-icon')
+        icon.dispatchEvent(new Event('click')); // To show Subcategories, so can be hidden on deslecting parent
```

Comment:
1. Add comment above the code
2. Fix spelling mistakes
3. Comment is not understandable. Elaborate it a bit more, or change the wordings.